### PR TITLE
feat(daemon): Critique Theater Phase 10 (adapter conformance lab + degraded registry)

### PR DESCRIPTION
**Pull request by Nagendhra-web**
<div align="center">

# 🎭 Critique Theater · Phase 10

**Adapter conformance lab and degraded registry.**

phase
files
tests
surface
stacked-on

</div>

> **TL;DR.** A nightly bench that runs every critique adapter against the same fixed transcripts, classifies the outcome, and writes provenly-broken providers into a short-lived skip registry so the orchestrator stops handing real users to known bad streams. Pure daemon-side, six new files, eighteen tests, zero edits to anything that ships on the request path today.

---

## 🤔 Why

Phase 6 shipped the parser. Phase 7 shipped the reducer. Both are pure, both are well-covered, both are right. What they cannot prove is that a **real provider stream**, arriving in whatever 512-byte network slices the network feels like producing today, still ends in a `ship` block with five panelists and a composite score.

That guarantee has to come from running each adapter end to end against a fixed transcript and measuring what falls out the other side. Until that bench exists, the only way to find out a provider has regressed is a real user hitting the failure first, which is exactly the surface Phase 15 is trying to flip on to broader rollout.

The adapter-degraded registry is the other half of the same loop. Once the bench can prove an adapter is broken, the orchestrator needs a cheap, in-memory record of "this adapter failed conformance, skip it for the next 24h" so the next user request does not get routed to the known-bad stream. Without the registry the bench is a green CI job nobody acts on; without the bench the registry is an empty data structure.

Phase 10 ships both halves of the loop so Phase 12 (observability) and the post-Phase-15 orchestrator wiring have the building blocks they need.

---

## 👀 What users will see

**On this PR:** nothing. No new public surface on the web layer, no new entry point in the project view, no flag flips. Pure daemon-side change behind the same internal critique API surface.

**After the post-Phase-15 orchestrator wiring lands:**

- A critique run that previously got routed to a regressed adapter now skips that adapter for the next 24h and routes through a healthy provider instead. The user sees a successful run where they would otherwise have seen a degraded banner or a failed run.
- The on-call dashboards (Phase 12) get a populated "adapters currently in degraded state" panel driven by the same registry this PR ships, so the team can spot a broken provider before users hit the failure.
- A provider that has been fixed since the last nightly conformance run gets retried on the next cycle without a manual `clearDegraded` call (the default TTL is intentionally short for this reason).

---

## 🎯 Surface area

- [x] Daemon (`apps/daemon`): new `critique/adapter-degraded.ts` registry, `critique/conformance.ts` harness, two synthetic adapters under `critique/__fixtures__/adapters/`, two new vitest files. Existing daemon code is untouched on the request path.
- [ ] Web runtime (`apps/web`)
- [ ] Contracts (`packages/contracts`)
- [ ] E2E (`e2e`)
- [ ] Docs (`docs/`, `README*`, `AGENTS.md`)
- [ ] Tools (`tools/dev`, `tools/pack`, `tools/pr`)
- [ ] Packaged runtime (`apps/packaged`, `apps/desktop`)
- [ ] Skills / design templates / design systems / craft

Internal-only on the daemon side. The conformance harness and registry are exported from `apps/daemon/src/critique/` but nothing in the orchestrator request path imports them yet; that wiring lands after #1320 (Phase 15) so the bench can be reviewed independently of the request-path change.

---

## 🧪 Specimen tracking

| Specimen file | Fixture | Expected outcome | Reason on degraded |
|---|---|---|---|
| `synthetic-good.ts` | `__fixtures__/v1/happy-3-rounds.txt` | `shipped` | n/a |
| `synthetic-bad.ts` | `__fixtures__/v1/malformed-unbalanced.txt` | `degraded` | `malformed_block` |
| (future) oversize | TBD adapter that pushes a `>32 KB` block | `degraded` | `oversize_block` |
| (future) missing artifact | adapter that ships without an artifact ref | `degraded` | `missing_artifact` |

Each specimen is streamed in **512-byte chunks**, not as one buffered blob, so the parser is exercised on chunk boundaries the same way a real `EventSource` would feed it.

---

## 🧱 Bench instruments

**Instrument 1.** `apps/daemon/src/critique/adapter-degraded.ts`

In-memory registry. `markDegraded(adapterId, reason, ttlMs, source)` records that an adapter produced a broken stream. `isDegraded(adapterId)` returns true while the record is unexpired. `__setDegradedClockForTests` is the only thing in the file that knows about time: production code reads `Date.now()` through that seam, tests advance it, and the registry decays without a sweeper thread.

Default TTL is **24 hours**. Long enough that a nightly conformance run is what populates the list, short enough that a fixed provider gets retried on the next cycle without a manual `clearDegraded` call.

**Instrument 2.** `apps/daemon/src/critique/conformance.ts`

```
  transcript stream  ─►  parseCritiqueStream  ─►  event collector
                                                       │
                                                       ▼
                                            ┌──────────────────┐
                                            │  classify(outcome) │
                                            └────────┬──────────┘
                                                     │
                  shipped ◄──── ship event seen ─────┤
                  degraded ◄─── parser_warning ──────┤   (marks adapter)
                  failed ◄──── no ship + no warn ────┘
```

The classifier is the single decision point. The same harness is meant to be called by:

- the nightly conformance CI job (populates `markDegraded` for the next 24h)
- the orchestrator pre-flight (reads `isDegraded` before routing a user to an adapter)
- ad-hoc debugging when a provider misbehaves in prod

**Instrument 3.** `apps/daemon/src/critique/__fixtures__/adapters/`

The two synthetic adapters. They are not registered in the adapter registry; they exist as test inputs to the harness.

---

## ✅ Validation

Commands run on the current head (`b94b6976`, post-main-merge) before pushing:

| Command | Result |
|---|---|
| `pnpm --filter @open-design/daemon typecheck` | clean |
| `pnpm --filter @open-design/daemon test tests/critique-conformance.test.ts` | 11 / 11 |
| `pnpm --filter @open-design/daemon test tests/critique-adapter-degraded.test.ts` | 7 / 7 |

```
[bench 10.1]  adapter-degraded                                7 / 7
              writes a record                          ok
              isDegraded true until TTL                ok
              isDegraded false after clock advance     ok
              clearDegraded forgets the record         ok
              listDegraded returns active records      ok
              second markDegraded overwrites           ok
              source label is preserved                ok

[bench 10.3]  conformance harness                            11 / 11
              shipped on a clean transcript            ok
              degraded on malformed_block              ok    (registry call observed)
              degraded on oversize_block               ok
              failed when no ship event arrives        ok
              failed when the adapter throws           ok
              (plus six harness-level assertions)
```

Web tests for the Theater interrupt restore that landed alongside this branch are also green on the same head; the relevant suites are `tests/components/Theater/CritiqueTheaterMount.test.tsx` and `tests/components/Theater/InterruptButton.test.tsx`. The dismissed CHANGES_REQUESTED rounds from @lefarcen and @Siri-Ray on the previous head are addressed on `2849403a` and unchanged by this main-merge.

---

## 📸 Screenshots

Intentionally none. No UI surface is touched: pure daemon-side primitives behind the existing internal critique API. The user-visible surface from Phase 15 + the post-Phase-15 orchestrator wiring (degraded adapters being skipped, the on-call dashboard panel for currently-degraded adapters) is where screenshots will land.

---

## ⛔ What this PR does not do

- **No orchestrator wiring.** The orchestrator does not yet consult `isDegraded` before routing. That edit is deliberately separate so the bench can land green and be reviewed independently of the request-path change.
- **No CI cron yet.** The nightly job that drives this harness on every adapter is a Phase 12 follow-on (see the Phase 12 note below). This PR ships the building block the cron will call.
- **No new public surface on the web layer.** Pure daemon-side change.

---

## 📅 A note on Phase 12

Phase 12 is the observability slice: Prometheus counters, structured logs under the `critique` namespace, and the Grafana dashboard JSON at `tools/dev/dashboards/critique.json`. It is deliberately not in this stack because it only earns its keep once the orchestrator actually routes traffic through the critique pipeline. Bumping counters from code paths that nothing reaches yet would land a green PR full of dead instrumentation.

Phase 12 lands after the orchestrator-wiring PR that follows #1320 (Phase 15). At that point the conformance bench from this PR becomes the input the nightly cron pumps into the dashboard, and the resolver from Phase 15 becomes the gate the structured logs annotate every span with.

---

## 🛡️ Risk surface

| Concern | Mitigation |
|---|---|
| Registry leaks under load | TTL bounds the working set; each entry holds 4 fields, no transcript blob. |
| Test-only clock seam reaching prod | `__setDegradedClockForTests` is exported, but production callers never touch it; the function is named to be obvious in a grep audit. |
| 24h default is wrong for some adapters | Caller can pass an explicit `ttlMs` to `markDegraded`. The default exists so the conformance harness has a sensible value, not because it is universal. |

---

## 🧮 Diff anatomy

```
 apps/daemon/src/critique/adapter-degraded.ts                  | +131
 apps/daemon/src/critique/conformance.ts                       | +138
 apps/daemon/src/critique/__fixtures__/adapters/synthetic-good |  +44
 apps/daemon/src/critique/__fixtures__/adapters/synthetic-bad  |  +46
 apps/daemon/tests/critique-adapter-degraded.test.ts           |  +93
 apps/daemon/tests/critique-conformance.test.ts                |  +93
```

Six new files. Zero modifications to anything that ships on the request path today.

---

## 🔗 Stacked on #1315 (which stacks on #1314)

```
main
  └─ #1314  Phase 8 (8 components + barrel)
       └─ #1315  Phase 9 (mount wrapper + native i18n)   ← mergeable on dcb8f55e
            └─ this PR  Phase 10 (conformance + degraded registry)
```

When #1315 merges the diff here collapses to just the Phase 10 additions.

### CODE REVIEW

File: apps/web/src/components/Theater/CritiqueTheaterMount.tsx [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+        console.warn('[critique-theater] interrupt request failed', err);
+      }
+    });
+
+    dispatch({
+      type: 'interrupted',
+      runId: state.runId,
+      bestRound: bestRoundOf(state),
+      composite: bestCompositeOf(state),
+    });
```
Comment (Siri-Ray) [inline apps/web/src/components/Theater/CritiqueTheaterMount.tsx]: This dispatch makes the UI enter the terminal `interrupted` phase before the daemon accepts the interrupt, and the fetch result is never inspected. `fetch` resolves for HTTP failures, so a 404/409 from `POST /api/projects/:projectId/critique/:runId/interrupt` still falls through to this local terminal event; even a rejected request is swallowed while the dispatch below already ran. Because the reducer treats `interrupted` as sticky, a real later `ship`/`failed` event for the same run is ignored, leaving the Theater showing an interruption that did not actually happen. Please wait for an accepted daemon response (for example `res.ok` / 202) before dispatching `interrupted`, and on failures clear `interruptPending` or surface an error while continuing to accept the daemon's real terminal SSE event.

File: apps/web/src/i18n/types.ts [hunk 2]
```diff
@@ -1469,4 +1469,45 @@ export interface Dict {
...
+  'critiqueTheater.degradedReasonMalformed': string;
+  'critiqueTheater.degradedReasonOversize': string;
+  'critiqueTheater.degradedReasonAdapter': string;
+  'critiqueTheater.degradedReasonProtocol': string;
+  'critiqueTheater.degradedReasonMissingArtifact': string;
+  'critiqueTheater.replay': string;
+  'critiqueTheater.replaySpeed': string;
+  'critiqueTheater.readOnly': string;
+  'critiqueTheater.shippedSummary': string;
+  'critiqueTheater.interruptedSummary': string;
```
Comment (Siri-Ray) [inline apps/web/src/i18n/types.ts:1607]: Adding `critiqueTheater.interruptedSummary` to the required `Dict` interface makes every locale object responsible for this key, but the PR only defines it in `en` and `zh-CN`; the newly changed `de`, `ja`, `ko`, and `zh-TW` blocks jump from `shippedSummary` directly to `shippedBadge`, and the untouched locale files still have none of the new Theater keys. Since each locale exports `const ...: Dict`, web typecheck will fail as soon as dependencies are installed. Please add this key, and the rest of the new `critiqueTheater.*` keys, to every `apps/web/src/i18n/locales/*.ts` dictionary or change the i18n contract so partial locale additions are explicitly allowed.

**Review by Siri-Ray — COMMENTED** [review]
Found two blocking issues in this pass: the new Theater i18n contract is not populated across the locale dictionaries, and the interrupt UI can locally terminalize a run even when the daemon rejects the interrupt. Details inline.


<sub>Generated by Looper 0.6.0 · runner=reviewer · agent=codex</sub>

**Review by lefarcen — COMMENTED** [review]
Hey @Nagendhra-web! 🎉

Solid architecture on the conformance bench and degraded registry — the clock seam is clean, TTL decay is sound, and the synthetic adapters are well-scoped. A few edge-case gaps and test coverage items to address:

**Lens A — Code correctness:**

Found 5 items (all P2 — should-fix edge cases and test gaps, no P1 blockers):

1. **Replay parsing lacks variant-level validation** (useCritiqueReplay.ts:236)
2. **SSE guard allows prototype pollution via role/__proto__** (state/sse.ts:61)
3. **Conformance classifier ignores parser_warning** (conformance.ts:76)
4. **Missing tests for oversize_block and adapter-throws** (conformance.test.ts)
5. **i18n incomplete: interrupted summary missing in de/ja/ko/zh-TW** (locales)

See inline comments for details.

**Lens B — Reasoning completeness:**

The PR body is thorough: 24h TTL rationale, specimen tracking table, risk surface mitigation. One clarification:

- The visible diff includes stacked Phase 7-9 Theater files because this targets `main`. Consider adding a note that the Phase 10-only delta is relative to #1315, or retarget the base while stacked.

No XSS found (React text/title rendering, no `dangerouslySetInnerHTML`). Once the inline items are addressed, this is good to ship.

Thanks for making open-design better! 🚀

File: apps/web/src/components/Theater/hooks/useCritiqueReplay.ts [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+  const [, setTick] = useReducer((n: number) => n + 1, 0);
+  const set = (next: ReplayMeta) => {
+    if (ref.current.status === next.status && ref.current.error === next.error) return;
+    ref.current = next;
+    setTick();
+  };
+  return [ref.current, set];
+}
+
+function parseTranscript(raw: string): PanelEvent[] {
```
Comment (lefarcen) [inline apps/web/src/components/Theater/hooks/useCritiqueReplay.ts:236]: **P2 — Replay parsing missing variant-level validation**

The replay parser uses `isPanelEvent` (checks `type` + non-empty `runId` only) before dispatching. A malformed transcript line like `{"type":"ship","runId":"r"}` or `panelist_dim` with invalid role can pass into the reducer and crash later rendering (`toFixed`, role lookup/prototype key paths).

**Fix:** Reuse/export the variant-level validator from `state/sse.ts` (or move a full `PanelEvent` validator into contracts), then parse replay lines through that same guard. This gives replay the same robustness as live SSE.

File: apps/web/src/components/Theater/state/sse.ts [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+ *      reducer never sees it.
+ */
+/** Per-variant required-fields validator. `isPanelEvent` from contracts only
+ *  checks `type` is known and `runId` is non-empty, so a frame like
+ *  `{ type: 'ship', runId: 'r' }` would slip through to the reducer with every
+ *  other field undefined and crash downstream code that calls
+ *  `final.composite.toFixed(1)`. This second-pass filter enforces the shape
+ *  of each variant before the action is dispatched (lefarcen + Siri-Ray +
+ *  codex P2 on PR #1314). */
+function hasValidVariantShape(event: PanelEvent): boolean {
```
Comment (lefarcen) [inline apps/web/src/components/Theater/state/sse.ts:61]: **P2 — SSE variant guard allows prototype pollution via role**

The variant guard checks `role`/`cast`/`status`/`reason` as generic strings. A malformed or compromised frame with `role: "__proto__"` passes `hasValidVariantShape`, then `reducer.ts:161` can read `Object.prototype` as the panelist view and throw on `current.dims.slice()`.

**Fix:** Validate `role` and `cast` against `PANELIST_ROLES` (runtime enum check), and validate `status`, `reason`, `cause`, and warning `kind` against their known contract sets before dispatch.

File: apps/daemon/src/critique/conformance.ts [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+      runId: params.runId,
+      adapter: params.adapterId,
+      parserMaxBlockBytes: params.parserMaxBlockBytes ?? 262_144,
+      projectId: params.projectId ?? 'conformance',
+      artifactId: params.artifactId ?? `conformance-${params.runId}`,
+      onArtifact: (payload) => {
+        shipPayload = payload;
+      },
+    })) {
+      events.push(event);
```
Comment (lefarcen) [inline apps/daemon/src/critique/conformance.ts:165]: **P2 — Conformance classifier inconsistent with PR body contract**

The PR body states: `parser_warning` should classify the run as degraded and mark the adapter. But the harness returns `shipped` as soon as it sees a `ship` event, even if earlier events included `parser_warning` (`unknown_role`, `score_clamped`, `composite_mismatch`, `duplicate_ship`).

**Fix:** Either track parser warnings and return a degraded outcome before success, or update the PR/body contract to state that shipped-with-warnings is acceptable and add tests for that policy.

File: apps/daemon/tests/critique-conformance.test.ts [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+beforeEach(() => {
+  now = 1_000_000;
+  __setDegradedClockForTests({ now: () => now });
+});
+afterEach(() => {
+  __setDegradedClockForTests(null);
+  __resetDegradedRegistryForTests();
+});
+
+describe('adapter conformance harness (Phase 10)', () => {
```
Comment (lefarcen) [inline apps/daemon/tests/critique-conformance.test.ts:37]: **P2 — Missing tests for claimed coverage: oversize_block and adapter-throws**

The PR body claims coverage for `oversize_block` and "adapter throws", but this file only tests happy, malformed fixture, TTL, no-ship, and ID threading.

**Fix:** Add tests for:
- An oversize block (>32KB artifact ref or body)
- An async source that throws mid-stream

Since `runAdapterConformance` is the new classifier boundary, pin the claimed degraded/failed behavior with explicit tests.

File: apps/web/src/i18n/locales/de.ts [hunk 7]
```diff
@@ -875,6 +875,46 @@ export const de: Dict = {
...
+  'critiqueTheater.degradedHeading': 'Jury für diesen Lauf offline',
+  'critiqueTheater.degradedReasonMalformed': 'Ungültige Jury-Ausgabe (Parser hat den Block abgelehnt).',
+  'critiqueTheater.degradedReasonOversize': 'Jury-Ausgabe hat das sichere Byte-Budget überschritten.',
+  'critiqueTheater.degradedReasonAdapter': 'Adapter {adapter} unterstützt das Critique-Protokoll nicht.',
+  'critiqueTheater.degradedReasonProtocol': 'Adapter {adapter} verwendet eine nicht unterstützte Protokollversion.',
+  'critiqueTheater.degradedReasonMissingArtifact': 'Der Lauf endete ohne ein finales Artefakt.',
+  'critiqueTheater.replay': 'Wiedergabe',
+  'critiqueTheater.replaySpeed': 'Wiedergabegeschwindigkeit',
+  'critiqueTheater.readOnly': 'Nur lesen',
+  'critiqueTheater.shippedSummary': 'In Runde {round} freigegeben · Gesamtscore {composite}',
```
Comment (lefarcen) [inline apps/web/src/i18n/locales/de.ts:949]: **P3 — i18n incomplete: interruptedSummary missing in de/ja/ko/zh-TW**

These locales add native `critiqueTheater.shippedSummary` but omit native `critiqueTheater.interruptedSummary`, so interrupted runs fall back to English despite the PR claiming native Phase 9 translations for de/ja/ko/zh-TW.

**Fix:** Add the missing key in those locale blocks or narrow the PR claim to "partial Phase 9 translations".

File: apps/daemon/src/critique/conformance.ts [hunk 8]
```diff
@@ -0,0 +1,207 @@
...
+      if (event.type === 'run_started') {
+        castRoles = event.cast;
+      } else if (event.type === 'panelist_close') {
+        closedRoles.add(event.role);
+      } else if (event.type === 'parser_warning') {
+        parserWarningSeen = true;
+      } else if (event.type === 'ship') {
+        // Tightening over "ship event seen = shipped": any earlier
+        // parser_warning makes the stream non-clean (lefarcen P2);
```
Comment (lefarcen) [inline apps/daemon/src/critique/conformance.ts]: **P2 — Parser warnings after first ship are missed.**

The loop returns as soon as `ship` is seen (line 156), so any `parser_warning` emitted *after* the first ship is ignored. Concrete failure: a duplicate `<SHIP>` block causes the parser to emit the first `ship` event, then later emit `parser_warning` kind `duplicate_ship`, but conformance already classified `shipped` and returned.

Fix: accumulate events until the stream ends (no early return on `ship`), then classify based on the full event set. A transcript with `ship` + any `parser_warning` should degrade, regardless of order.

File: apps/daemon/src/critique/conformance.ts [hunk 9]
```diff
@@ -0,0 +1,207 @@
...
+ */
+function toContractReason(r: ConformanceDegradedReason): DegradedReason {
+  switch (r) {
+    case 'parser_warning':   return 'malformed_block';
+    case 'incomplete_panel': return 'missing_artifact';
+    default:                 return r;
+  }
+}
+
+/**
```
Comment (lefarcen) [inline apps/daemon/src/critique/conformance.ts:125]: **P2 — closedRoles is cumulative across all rounds, not per-round.**

A shipped run requires all five cast roles closed *in the final shipping round*. The current code accumulates `closedRoles` across the entire transcript (line 119 adds every `panelist_close`), so a later round that omits some roles can still pass the cast-complete check (line 148) if those roles closed in an earlier non-shipping round.

Fix: reset `closedRoles` on each `round_started`, or track per-round and validate the shipping round's cast specifically.

**Review by lefarcen — COMMENTED** [review]
Hey @Nagendhra-web — great fixes on this pass! 🎉

Siri-Ray's P1 blockers are resolved:
- ✅ i18n now complete across all six locales (en/zh-CN/de/ja/ko/zh-TW)
- ✅ Interrupt race fixed — dispatches `interrupted` only after `res.ok`

Two P2 edge cases remain in the conformance classifier (see inline). Once those are tightened, this is ready to ship.

File: apps/daemon/src/critique/conformance.ts (hunk 8 above)
Comment (Nagendhra-web) [reply apps/daemon/src/critique/conformance.ts]: Fixed in a965849. The harness now drains the whole stream before classifying: `shipEvent` is captured on the first SHIP but the loop keeps iterating, so a later `parser_warning` (notably the `duplicate_ship` kind the parser emits when it sees a second `<SHIP>` block) still flips the run to degraded. The new test `classifies a duplicate-SHIP stream as degraded parser_warning` drives exactly the scenario you described and asserts the events array holds both the first ship AND the duplicate_ship warning so debuggers can see what happened.

**Review by lefarcen — COMMENTED** [review]
Great work on the fixes! 🎉

Both P2 edge cases from my prior review are now resolved:
- ✅ Parser warnings after first ship are caught (no early return; stream drains fully)
- ✅ closedRoles tracked per-round (not cumulative; validates the shipping round specifically)

No new issues found. Clean pass on this iteration.

**Review by Siri-Ray — DISMISSED** [review]
@Nagendhra-web I reviewed the current head across the changed Theater surface, the daemon conformance harness, degraded-adapter registry, locale additions, and the follow-up fixes for prior review threads. The conformance classifier now drains through post-ship warnings, validates the shipping round's closed cast per round, preserves the shipped artifact payload for callers, and the UI-side interrupt/replay/SSE guards address the previously reported correctness issues. I did not find any remaining actionable issues in the scoped changed ranges. Thanks for the careful follow-through on this stack!

Note: I attempted to run the focused daemon and web Theater Vitest suites, but this checkout has no installed node_modules, so vitest was unavailable locally; I relied on diff review plus the reported green checks.


<sub>🔁 Powered by <a href="https://github.com/nexu-io/looper">Looper</a> · runner=reviewer · agent=opencode · An autonomous AI dev team for your GitHub repos.</sub>

**Review by Siri-Ray — DISMISSED** [review]
@Nagendhra-web I reviewed the current head across the daemon conformance harness, degraded-adapter registry, synthetic fixture adapters, Theater interrupt/replay/SSE guard follow-ups, locale additions, and the related focused tests in the changed ranges. The prior conformance edge cases are now covered by draining through post-ship warnings and validating panelist closes per shipping round, the shipped outcome surfaces the parsed artifact payload, and the web-side replay/SSE/interrupt fixes line up with the reducer behavior. I did not find any remaining actionable issues in this pass. Thanks for the careful follow-through on this stack!

Note: I attempted the focused daemon and web Theater Vitest commands locally, but this checkout does not have `vitest` available through `pnpm exec`; at review time GitHub showed the latest build/workspace checks still pending.


<sub>🔁 Powered by <a href="https://github.com/nexu-io/looper">Looper</a> · runner=reviewer · agent=opencode · An autonomous AI dev team for your GitHub repos.</sub>

**Review by lefarcen — COMMENTED** [review]
Hey @Nagendhra-web 👋

Nice follow-through on this stack — the conformance harness fixes are holding up, and the merge-main refresh also keeps the Theater reset paths aligned with the newer UI surface. ✨

I re-reviewed the current head (`9bd5749`) across the daemon conformance/registry code, the Theater SSE/replay/reducer paths, and the locale/test follow-ups. The earlier edge cases remain addressed: post-ship parser warnings are still caught, closed roles are scoped to the shipping round, malformed replay/SSE variants are filtered before reducer dispatch, and interrupt terminalization waits for a successful daemon response. I did not find any new actionable issues in this pass.

I could not run the focused Vitest suites locally because this checkout has no installed `node_modules`, so this is a diff review plus the reported/CI signal rather than a local test run. CI is still pending on this head, so I’m leaving this as a clean COMMENT rather than an approval. Looks good from my side once the remaining checks and human merge gate clear. ❤️

File: apps/web/src/components/Theater/CritiqueTheaterMount.tsx [hunk 10]
```diff
@@ -0,0 +1,143 @@
...
+        typeof process !== 'undefined'
+        && process.env?.NODE_ENV === 'development'
+      ) {
+        // eslint-disable-next-line no-console
+        console.warn('[critique-theater] interrupt request failed', err);
+      }
+    });
+
+    const { round, composite } = bestRoundAndComposite(state);
+    dispatch({
```
Comment (lefarcen) [inline apps/web/src/components/Theater/CritiqueTheaterMount.tsx]: P1 This reintroduces the interrupt race: the UI dispatches the sticky `interrupted` terminal event before the daemon accepts the POST, while `fetch` failures are only logged and HTTP 404/409 still resolve. If the daemon rejects the interrupt or the run has already shipped/failed, the reducer will ignore the real terminal SSE event and show an interruption that never happened. Please snapshot `runId`/round/composite at click time, wait for the POST response, dispatch `interrupted` only on an accepted response (e.g. `res.ok` / 204), and clear `interruptPending` or surface an error on rejection/non-2xx so the daemon's real terminal event can still win.

**Review by lefarcen — CHANGES_REQUESTED** [review]
Hey @Nagendhra-web — thanks for keeping this branch rebased with the Phase 8 landing. The stricter contract-side `PanelEvent` guard is a good direction and keeps the SSE/replay validation centralized.

One previously fixed interrupt correctness issue appears to have regressed in this push. It is small and localized, but it needs restoring before this can merge because it can leave the Theater in a terminal state the daemon never accepted. See the inline note.

Once that handler waits for the daemon acknowledgement again, this should be back on track.

File: apps/web/src/components/Theater/CritiqueTheaterMount.tsx (hunk 10 above)
Comment (Siri-Ray) [inline apps/web/src/components/Theater/CritiqueTheaterMount.tsx]: This reintroduces the interrupt race on the current head: the handler dispatches the sticky `interrupted` terminal event immediately after starting the POST, while the `fetcher(...).catch(...)` above only logs network failures and never checks non-2xx HTTP responses. If the daemon rejects the interrupt, the endpoint is missing, or the run already shipped/failed, the reducer will still move to `interrupted` and then ignore the daemon's real terminal SSE event. Please snapshot the clicked `runId`/best round/composite, wait for the POST to resolve, dispatch `interrupted` only when the daemon accepts it (for example `res.ok`/204), and clear `interruptPending` or surface the failure on rejection/non-2xx so the real daemon terminal event can still win.

<sub>🔁 Powered by <a href="https://github.com/nexu-io/looper">Looper</a> · runner=reviewer · agent=opencode · An autonomous AI dev team for your GitHub repos.</sub>

**Review by Siri-Ray — CHANGES_REQUESTED** [review]
@Nagendhra-web thanks for continuing to push this stack forward and for the conformance harness cleanup. I found one current-head regression in the Theater interrupt path that can put the UI into a terminal state the daemon did not accept, so I’m requesting changes on that localized issue.


<sub>🔁 Powered by <a href="https://github.com/nexu-io/looper">Looper</a> · runner=reviewer · agent=opencode · An autonomous AI dev team for your GitHub repos.</sub>

**Review by Siri-Ray — DISMISSED** [review]
@Nagendhra-web I reviewed the current head across the adapter degraded registry, conformance harness and synthetic fixtures, the focused daemon coverage, the Theater interrupt follow-up, SSE validation path, locale additions, and the related web tests. The latest interrupt handler now waits for a successful daemon acknowledgement before terminalizing, the conformance harness drains through post-ship warnings and validates the shipping round independently, and the degraded registry TTL behavior is covered without introducing request-path side effects. I did not find any remaining actionable issues in the scoped changed ranges. Thanks for the careful iteration on this stack!


<sub>🔁 Powered by <a href="https://github.com/nexu-io/looper">Looper</a> · runner=reviewer · agent=opencode · An autonomous AI dev team for your GitHub repos.</sub>

**Review by lefarcen — DISMISSED** [review]
Hey @Nagendhra-web — nice recovery on this round.

I re-checked the current head and the interrupt regression from my previous review is fixed: the handler snapshots the clicked run values, waits for the daemon response, only terminalizes on an accepted response, and clears the pending state on rejection/non-2xx so the real SSE terminal event can still win. The added negative tests cover the rejection and 404 paths, which is exactly the gap that regressed.

That clears my prior blocking review. Thanks for the careful iteration here — this is clean from my side now. ✨

**Review by Siri-Ray — DISMISSED** [review]
@Nagendhra-web I reviewed the current head across the scoped daemon changes: the conformance harness now drains through post-ship parser warnings, validates completeness against the shipping round, returns the captured ship artifact payload, and the synthetic fixture adapters resolve their transcript files from module-relative URLs. The added conformance tests cover the prior warning, incomplete-panel, oversize, thrown-adapter, and artifact assertions, and I did not find any actionable issues in the changed ranges. Thanks for the careful follow-through on this stack.

Note: I could not run the focused Vitest suite locally because this checkout has no installed `node_modules`. GitHub's `build` check is passing; `Validate workspace` is currently failing in `apps/web/tests/components/FileViewer.test.tsx`, which is outside this PR's current changed ranges.


<sub>🔁 Powered by <a href="https://github.com/nexu-io/looper">Looper</a> · runner=reviewer · agent=codex · An autonomous AI dev team for your GitHub repos.</sub>

**Review by lefarcen — COMMENTED** [review]
Hey @Nagendhra-web — thanks for cleaning up the conflict resolution and documenting exactly what was kept from the Phase 10 branch. I rechecked the current PR file list and the four daemon conflict-resolution files; the conformance harness shape still looks consistent with the previously approved pass, and I do not have new inline code findings on this head.

Two bits to clean up before this can move through the formal path:

- GitHub Actions currently has `Validate workspace` failing in `@open-design/web` typecheck: `tests/components/FileViewer.test.tsx(434,8)` is missing the required `projectKind` prop for `FileViewer`. It is outside the four current PR files, but this head still needs green checks before merge; if it is a stale merge-ref issue, a rerun after the latest sync should prove that.
- The repo now has the PR template from #1520, and this PR body is missing **Why**, **What users will see**, **Surface area**, and **Validation** in the new format. Since this is a feature-labeled Theater/daemon change, filling those in will help product/release review understand the user-visible surface. If you are using an AI assistant, this prompt should be enough to update it:

> Read the current PR body for nexu-io/open-design#1316 and the template at
> `.github/pull_request_template.md`. Rewrite the description filling in:
> - **Why**: use case + pain being addressed (not a one-line restatement)
> - **What users will see**: user perspective — what they click, what new
>   thing appears, what default changes
> - **Surface area**: check every applicable box; pick "None" if internal-only
> - **Screenshots** (if UI is checked): entry-point screenshot, before/after
> - **Validation**: what you actually ran (`pnpm guard`, `pnpm typecheck`, etc.)
> Then update via `gh pr edit 1316 --repo nexu-io/open-design --body-file -`.

No new code blockers from my pass on the daemon conflict-resolution files; this is mainly checks + PR metadata hygiene now. 🙌

**Review by Siri-Ray — APPROVED** [review]
@Nagendhra-web I reviewed the current head across the scoped changed ranges: the daemon conformance harness now drains through post-ship parser warnings, validates shipping-round panel completeness, preserves the captured artifact payload on shipped outcomes, and the synthetic adapters resolve fixtures from module-relative URLs. I also checked the FileViewer test fix for the new projectKind prop and confirmed the live GitHub checks are passing. I could not run the focused Vitest suites locally because this worktree does not have node_modules installed, but I did not find any actionable issues in the diff. Thanks for the careful follow-through on this stack. 🙂


<sub>🔁 Powered by <a href="https://github.com/nexu-io/looper">Looper</a> · runner=reviewer · agent=codex · An autonomous AI dev team for your GitHub repos.</sub>

**Review by lefarcen — COMMENTED** [review]
Hey @Nagendhra-web — nice follow-up on the workspace failure cleanup. I rechecked the current head across the daemon conformance files and the focused `FileViewer` test update. The harness still preserves the important fixes from the prior pass: it drains through post-ship warnings, validates closed panelists per shipping round, returns the captured ship artifact payload, and keeps the synthetic fixture paths module-relative. The `projectKind="prototype"` addition also addresses the typecheck failure I called out on the previous head.

I do not have new inline code findings on this push. The only housekeeping item still visible from my side is the PR-body template update noted in the previous review, so reviewers/product can quickly scan Why / user-visible surface / validation. 🙌
